app/src/main/java/com/cakii/nimble_android_challenge/data/interceptor/AuthInterceptor.kt

```diff
@@ -0,0 +1,46 @@
+package com.cakii.nimble_android_challenge.data.interceptor
+
+import android.content.Context
+import com.cakii.nimble_android_challenge.R
+import com.cakii.nimble_android_challenge.data.entity.Auth
+import com.cakii.nimble_android_challenge.data.entity.AuthRequest
+import com.cakii.nimble_android_challenge.utils.Prefs
+import com.google.gson.Gson
+import okhttp3.*
+import java.io.IOException
+import java.net.URL
+
+
+class AuthInterceptor(private val context: Context, private val prefs: Prefs): Authenticator {
```

Comment:
this extends from an `Authenticator` 😇 so it should be `AuthAuthenticator`, not Interceptor